Assets/MixedRealityToolkit/_Core/Boundary/MixedRealityBoundarySystem.cs

```diff
@@ -11,45 +15,112 @@ namespace Microsoft.MixedReality.Toolkit.Internal.Managers
+            base.Initialize();
+            InitializeInternal();
+        }
+
+        /// <summary>
+        /// Performs initialization tasks for the BoundaryManager.
+        /// </summary>
+        private void InitializeInternal()
+        {
+            XRDevice.SetTrackingSpaceType(TrackingSpaceType);
+
+            CalculateBoundaryBounds();
+            SetPlatformBoundaryVisibility();
         }
 
         /// <summary>
         /// Optional ProfileUpdate function to allow reconfiguration when the active configuration profile of the Mixed Reality Manager is replaced
         /// </summary>
         public override void Reset()
         {
-            // TODO React to profile change
+            base.Reset();
+            InitializeInternal();
         }
 
         /// <summary>
-        /// Optional Update function to perform per-frame updates of the manager
+        /// Retrieves the boundary geometry and creates the boundary and inscribed playspace volumes.
         /// </summary>
-        public override void Update()
+        private void CalculateBoundaryBounds()
         {
-            // TODO Update stuff 
+            if (!Boundary.configured)
+            {
+                // The user has not configured a playspace boundary on this device.
+                return;
+            }
+
+            if (XRDevice.GetTrackingSpaceType() != TrackingSpaceType.RoomScale)
+            {
+                Debug.Log("Boundaries are supported for Room Scale experiences only.");
+                return;
+            }
+
+            BoundaryVolume = new Bounds();
+
+            // Get the boundary geometry.
+            List<Vector3> boundaryGeometry = new List<Vector3>(0);
+            if (Boundary.TryGetGeometry(boundaryGeometry))
+            {
+                for (int i = 0; i < boundaryGeometry.Count; i++)
+                {
+                    BoundaryVolume.Encapsulate(boundaryGeometry[i]);
+                }
+
+                // todo: CreateInscribedBounds()
+
+                // Set the "ceiling" of the space using the configured height.
+                BoundaryVolume.Encapsulate(new Vector3(0f, BoundaryHeight, 0f));
+            }
         }
 
         /// <summary>
-        /// Optional Destroy function to perform cleanup of the manager before the Mixed Reality Manager is destroyed
+        /// Sets the property indicating if the boundary should be rendered by the platform.
         /// </summary>
-        public override void Destroy()
+        /// <remarks>
+        /// Not all platforms support specifying whether or not to render the playspace boundary.
+        /// For platforms without boundary rendering control, the default behavior will be unchanged 
+        /// regardless of the value provided.
+        /// </remarks>
+        private void SetPlatformBoundaryVisibility()
         {
-            // TODO Destroy stuff 
+            if (Boundary.configured)
+            {
+#if !UNITY_WSA
```

Comment:
We should probably avoid platform specific calls in our profiles if we can.

Reply:
Agreed. Just not sure what the correct path is for platforms that lack specific features... Is it Unity's job our ours to handle these types of things.

Reply:
If we are providing a boundary mechanism. Then we need to provide a facility to mock it, or at least not break if the platform doesn't provide it